typescript/cli/src/utils/output.ts

```diff
@@ -0,0 +1,156 @@
+import chalk from 'chalk';
+
+import { CheckerViolation } from '@hyperlane-xyz/sdk';
+import { deepEquals, isNullish, isObject } from '@hyperlane-xyz/utils';
+
+interface ViolationOutput
+  extends Pick<CheckerViolation, 'actual' | 'expected'> {}
+
+export type ViolationDiff =
+  | {
+      [key: string]: ViolationOutput | ViolationDiff;
+    }
+  | ViolationDiff[];
+
+/**
+ * Takes 2 values and sorts them by the `type` field if they have it.
+ * This helps when comparing `IsmConfig` or `HookConfig` arrays
+ */
+export function sortByType(a: any, b: any): number {
+  if (a.type < b.type) {
+    return -1;
+  }
+
+  if (a.type > b.type) {
+    return 1;
+  }
+
+  return 0;
```

Comment:
as long as `a` and `b` are not undefined it shouldn't throw but it will if they are. Considering that this func is used only when 2 arrays have the same length it shouldn't happen but better be safe than sorry I'm going to cover this scenario.